rust/src/utils.rs

```diff
@@ -59,11 +91,27 @@ macro_rules! to_bytes {
     }
 }
 
+#[macro_export]
+macro_rules! to_hex {
+    ($name:ident) => {
+        #[wasm_bindgen]
+        impl $name {
+            pub fn to_hex(&self) -> String {
+                let mut buf = Serializer::new_vec();
+                self.serialize(&mut buf).unwrap();
+                hex::encode(buf.finalize())
+            }
+        }
+    }
+}
+
 #[macro_export]
 macro_rules! to_from_bytes {
     ($name:ident) => {
         to_bytes!($name);
         from_bytes!($name);
+        to_hex!($name);
+        from_hex!($name);
```

Comment:
@alessandrokonrad Which format should we be using here for hex? The raw bytes? or the CBOR bytes? Currently `from_bytes()` can return one of those depending on which types. Most types are CBOR bytes, but Addresses, crypto keys and hashes are all raw bytes. See: #263

What is your thoughts on this? I proposed having `to_raw_bytes()` and `to_cbor_bytes()` but then if we add in hex to this too that's a lot of variants.

Reply:
> Which format should we be using here for hex? The raw bytes? or the CBOR bytes?

to_hex/from_hex are simply wrappers around to_bytes/from_bytes. So it returns/takes whatever that is. The macro to_from_bytes is implemented for types that are fully in CBOR I think. I'm not sure how these would look like in raw bytes, for instance `Value`?

> Most types are CBOR bytes, but Addresses, crypto keys and hashes are all raw bytes

I think it makes sense to return them in a format how the cardano ledger would take them, so probably as it is right now. Everything else would be more confusing

Reply:
>Would to_cbor_bytes simply add a tag in front of the raw bytes, that says something is bytes?

In the cases of the hashes/keys and addresses, yes. They're all defined in the .cddl as `address = bytes` (usually with size limits) for example. When they're used in the context of the CDDL CBOR encoding (e.g. when composed into the other CBOR structures in that spec) they need to have those extra bytes. This is related to our discussion in cardano-foundation/CIPs#148 which going by how the spec is written would need those bytes, which is something that we might want to change there. Either way it's a bit confusing IMO having these two notions of what a byte representation means and I'm not sure about the best way to go about it, which is why I opened up #263.

>I think it makes sense to return them in a format how the cardano ledger would take them, so probably as it is right now

If we stay with this we should at least document it.

Reply:
> This is related to our discussion in cardano-foundation/CIPs#148 which going by how the spec is written would need those bytes, which is something that we might want to change there

Yes I would be for a change there. Or we add a simple helper function in either the message signing library or the serialization-lib to add a tag in front of the raw bytes.